src/LibraryViewExtension/Handlers/SearchResultDataProvider.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using CefSharp;
+using Dynamo.Search;
+using Dynamo.Search.SearchElements;
+
+namespace Dynamo.LibraryUI.Handlers
+{
+    class LoadedTypeItemExtended : LoadedTypeItem
+    {
+        public int weight { get; set; }
+    }
+
+    /// <summary>
+    /// Provides json resource data for all the loaded nodes
+    /// </summary>
+    class SearchResultDataProvider : NodeItemDataProvider
+    {
+        private const string serviceIdentifier = "search";
+        private IEnumerable<NodeSearchElement> elements;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="items"></param>
+        public SearchResultDataProvider(NodeSearchModel model) : base(model)
+        {
+        }
+
+        public override Stream GetResource(IRequest request, out string extension)
+        {
+            var url = request.Url;
+            var uri = new Uri(url);
+            var pathAndQuery = uri.PathAndQuery;
+            var index = url.IndexOf(serviceIdentifier);
+            var text = url.Substring(index + serviceIdentifier.Length + 1);
```

Comment:
For the second point, then you need to override GetLoadedTypeDataForRequest. But as GetResource is only a function with few lines.